catalyst-gateway/bin/src/event_db/follower.rs

```diff
@@ -46,11 +46,106 @@ impl EventDB {
         Ok(())
     }
 
+    /// Get current slot info for the provided date-time and network
+    pub(crate) async fn current_slot_info(
+        &self, date_time: DateTime, network: Network,
+    ) -> Result<(SlotNumber, BlockHash, DateTime), Error> {
+        let conn = self.pool.get().await?;
+
+        let network = match network {
```

Comment:
Can we consider extending `Network` from `cardano_chain_follower` by adding `as_str()` like so?

```rust
trait StringRepresentable {
    fn as_str(&self) -> String;
}

impl StringRepresentable for Network {
    fn as_str(&self) -> String {
        match self {
            Network::Mainnet => "mainnet".to_string(),
            Network::Preview => "preview".to_string(),
            Network::Preprod => "preprod".to_string(),
            Network::Testnet => "testnet".to_string(),
        }
    }
}
```

Then, here and elsewhere, we can do:

```rust
        let network = match network {
            Network::Mainnet => Network::Mainnet.as_str(),
            Network::Preview => Network::Preview.as_str(),
            Network::Preprod => Network::Preprod.as_str(),
            Network::Testnet => Network::Testnet.as_str(),
        }; 
```

Reply:
if Network implements `as_str()` can't this just then be simplified to:
```
let network = network.as_str();
```

Reply:
we have [something similar ](https://github.com/input-output-hk/hermes/blob/main/hermes/crates/cardano-chain-follower/src/lib.rs#L114)

Reply:
agree, seems it's time to update `chain-follower` crate 😊